init-interfaces.sh

```diff
@@ -0,0 +1,107 @@
+    return 0 # true
+  fi
+  return 1 # false
+}
+
+get_device_by_mac() {
+  local given_mac=$1
+  local device_name
+
+  for dev in $(nmcli device status | awk '/ethernet/ {print $1}'); do
+    dev_mac=$(nmcli -g GENERAL.HWADDR dev show "$dev" | sed -e 's/\\//g' | tr '[:upper:]' '[:lower:]')
+    if [[ "$dev_mac" == "$given_mac" ]]; then
+      device_name="$dev"
+      break
+    fi
+  done
+  echo "$device_name"
+}
+
+get_con_uuid_by_device_mac() {
+  local given_mac=$1
+  local con_name
+  local con_uuid
+  local mac_dev_name
+
+  mac_dev_name=$(get_device_by_mac "$given_mac")
+  IFS=$'\n'; for con_name in $(nmcli -t -f NAME con show); do
+    con_dev_name=$(nmcli -g connection.interface-name con show "$con_name")
+    if [[ "$con_dev_name" == "$mac_dev_name" ]]; then
+      con_uuid=$(nmcli -g CONNECTION.UUID con show "$con_name")
+      break
+    fi
+  done
+  echo "$con_uuid"
+}
+
+if [[ ! -f /boot/mac_addresses ]] ; then
+  echo "no mac address configuration file found .. exiting"
+  exit 1
+fi
+
+primary_mac="$(awk -F= '/PRIMARY_MAC/ {print $2}' < /boot/mac_addresses | tr '[:upper:]' '[:lower:]')"
+secondary_mac="$(awk -F= '/SECONDARY_MAC/ {print $2}' < /boot/mac_addresses | tr '[:upper:]' '[:lower:]')"
+
+default_device=""
+secondary_device=""
+default_connection_uuid=""
+secondary_connection_uuid=""
+
+for dev in $(nmcli device status | awk '/ethernet/ {print $1}'); do
+  dev_mac=$(nmcli -g GENERAL.HWADDR dev show "$dev" | sed -e 's/\\//g' | tr '[:upper:]' '[:lower:]')
+  case $dev_mac in
+    $primary_mac)
+      default_device="$dev"
+      default_connection_uuid=$(get_con_uuid_by_device_mac "$dev_mac")
+      ;;
+    $secondary_mac)
+      secondary_device="$dev"
+      secondary_connection_uuid=$(get_con_uuid_by_device_mac "$dev_mac")
+      ;;
+    *)
+      ;;
+   esac
+done
+
+echo -e "default dev: $default_device (CONNECTION.UUID $default_connection_uuid)\nsecondary dev: $secondary_device (CONNECTION.UUID $secondary_connection_uuid)"
+if [[ -z "$default_device" ]] || [[ -z "$secondary_device" ]]; then
+  echo "error: primary/secondary device name not found"
+  exit 1
+fi
+
+if eval ! is_con_exists_by_uuid "$default_connection_uuid"; then
+  nmcli con add type ethernet \
+                conn.interface "$default_device" \
+                connection.autoconnect yes \
+                ipv4.method auto
+  default_connection_uuid=$(get_con_uuid_by_device_mac "$primary_mac")
+fi
+if eval ! is_con_active_by_uuid "$default_connection_uuid"; then
+  nmcli con up uuid "$default_connection_uuid"
+fi
+
+if eval ! is_con_exists_by_uuid "$secondary_connection_uuid"; then
+  nmcli con add type ethernet \
+                conn.interface "$secondary_device" \
+                connection.autoconnect no \
+                ipv4.method disabled
+  secondary_connection_uuid=$(get_con_uuid_by_device_mac "$secondary_mac")
+fi
+if eval is_con_active_by_uuid "$secondary_connection_uuid"; then
+  nmcli con down uuid "$secondary_connection_uuid"
+fi
```

Comment:
Since UUID can be duplicated it can happend that we don't add the connection when needed or we put down something that is not suppose to be down.